### PR TITLE
test/cql-pytest/test_tools.py: improve tempdir usage for scrub tests

### PR DESCRIPTION
**Pull request by denesb**
Scrub tests use a lot of temporary directories. This is suspected to cause problems in some cases. To improve the situation, this patch:
* Creates a single root temporary directory for all scrub tests
* All further fixtures create their files/directories inside this root dir.
* All scrub tests create their temporary directories within this root dir.
* All temporary directories now use an appropriate "prefix", so we can tell which temporary directory is part of the problem if a test fails.

Refs: #14309

### CODE REVIEW

File: test/cql-pytest/test_tools.py [hunk 1]
```diff
@@ -696,18 +696,26 @@ def test_external_dir_autodetect_conf_dir_home_env(self, scylla_path, system_scy
 
 
 @pytest.fixture(scope="module")
-def scrub_schema_file():
```
Comment (bhalevy) [inline test/cql-pytest/test_tools.py:699]: Wouldn't it be enough to change to `scope="function"`?

Comment (denesb) [reply test/cql-pytest/test_tools.py:699]: We don't want to regenerate this file on every test, slowing down all tests. This file is read-only, once it is generated.

Comment (bhalevy) [reply test/cql-pytest/test_tools.py:699]: Ah, makes sense

**Review by nyh — APPROVED** [review]
Looks good to me, but I think maybe the commit message is misleading (or I misunderstood it). You say this patch is to "improve the situation" but if I understand correctly, it just helps to debug it - you have better-named directories and they are organized inside a parent directory. If I understand correctly, you still have as many directories as you had before, right?

**Comment by denesb** [comment]
Yes, I have just as many temporary directories as before. They are just organized and named better.

**Comment by denesb** [comment]
v2:
* Removed unncessary usage of a fixture.

**Review by bhalevy — APPROVED** [review]

**Comment by denesb** [comment]
@scylladb/scylla-maint ping.

**Pull request closed without merging**